jkube-kit/generator/webapp/src/test/java/org/eclipse/jkube/generator/webapp/handler/WildFlyAppSeverHandlerTest.java

```diff
@@ -25,74 +27,77 @@
-import org.junit.Rule;
-import org.junit.Test;
-import org.junit.rules.TemporaryFolder;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
 
-import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.assertFalse;
-import static org.junit.Assert.assertTrue;
+import static org.assertj.core.api.Assertions.assertThat;
 
-public class WildFlyAppSeverHandlerTest {
+@SuppressWarnings({"unused", "ResultOfMethodCallIgnored"})
+class WildFlyAppSeverHandlerTest {
 
-  @Rule
-  public TemporaryFolder temporaryFolder = new TemporaryFolder();
+  @TempDir
+  Path temporaryFolder;
 
   @Mocked
   private GeneratorContext generatorContext;
 
   @Test
-  public void isApplicableHasJmsXmlShouldReturnTrue() throws IOException {
+  void isApplicableHasJmsXmlShouldReturnTrue() throws IOException {
     // Given
     // @formatter:off
     new Expectations() {{
-      generatorContext.getProject().getBuildDirectory(); result = temporaryFolder.getRoot();
+      generatorContext.getProject().getBuildDirectory(); result = temporaryFolder.toFile();
     }};
     // @formatter:on
-    assertTrue(new File(temporaryFolder.newFolder("META-INF"), "some-file-with-jms.xml").createNewFile());
-    assertTrue(new File(temporaryFolder.newFolder("META-INF-1337"), "context.xml").createNewFile());
+    File baseDir = Files.createDirectory(temporaryFolder.resolve("META-INF")).toFile();
+    assertThat(new File(baseDir, "some-file-with-jms.xml").createNewFile()).isTrue();
+    baseDir = Files.createDirectory(temporaryFolder.resolve("META-INF-1337")).toFile();
+    assertThat(new File(baseDir, "context.xml").createNewFile()).isTrue();
     // When
     final boolean result = new WildFlyAppSeverHandler(generatorContext).isApplicable();
     // Then
-    assertTrue(result);
+    assertThat(result).isTrue();
   }
 
   @Test
-  public void isApplicableHasJmsXmlAndThorntailPluginShouldReturnFalse(@Mocked Plugin plugin) throws IOException {
+  void isApplicableHasJmsXmlAndThorntailPluginShouldReturnFalse(@Mocked Plugin plugin) throws IOException {
     // Given
     // @formatter:off
     new Expectations() {{
-      generatorContext.getProject().getBuildDirectory(); result = temporaryFolder.getRoot(); minTimes = 0;
+      generatorContext.getProject().getBuildDirectory(); result = temporaryFolder; minTimes = 0;
       plugin.getGroupId(); result = "io.thorntail";
       plugin.getArtifactId(); result = "thorntail-maven-plugin";
       generatorContext.getProject().getPlugins(); result = Collections.singletonList(plugin);
     }};
     // @formatter:on
-    assertTrue(new File(temporaryFolder.newFolder("META-INF"), "some-file-with-jms.xml").createNewFile());
-    assertTrue(new File(temporaryFolder.newFolder("META-INF-1337"), "context.xml").createNewFile());
+    File baseDir = Files.createDirectory(temporaryFolder.resolve("META-INF")).toFile();
+    assertThat(new File(baseDir, "some-file-with-jms.xml").createNewFile()).isTrue();
```

Comment:
As commented [here](https://github.com/eclipse/jkube/issues/1563#issuecomment-1203719846), these assertions are not necessary if you use the nio Files utility.